src/Data/RDF/State.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.RDF.State where
+
+import           Control.Monad.State
+import           Data.RDF            (RDF, Rdf, PrefixMappings, BaseUrl, Triples, Triple, NodeSelector)
+import qualified Data.RDF            as RDF
+
+-- | StateT monad which allows to combine more easily functions from 'Rdf'
+-- typeclass, eg.:
+--
+-- @
+-- import Data.RDF
+-- import qualified Data.RDF.State as RDFS
+--
+-- main :: IO ()
+-- main = do
+--   let myEmptyGraph = empty :: RDF TList
+--   newGraph <- execStateT createGraph myEmptyGraph
+--   putStrLn (showGraph newGraph)
+--
+-- createGraph :: (Rdf rdfImpl, Monad m) => RdfST rdfImpl m ()
+-- createGraph = do
+--   -- add a triple to the empty graph
+--   let triple1 = triple (unode "http://www.example.com/rob")
+--                        (unode "http://xmlns.com/foaf/0.1/interest")
+--                        (unode "http://dbpedia.org/resource/Scotch_whisky")
+--   RDFS.addTriple triple1
+--
+--   -- add another triple to the graph
+--   let triple2 = triple (unode "http://www.example.com/rob")
+--                        (unode "http://xmlns.com/foaf/0.1/interest")
+--                        (unode "http://dbpedia.org/resource/Haskell_(programming_language)")
+--   RDFS.addTriple triple2
+--
+--   -- remove one of my interests
+--   RDFS.removeTriple triple1
+-- @
+type RdfST rdfImpl m a = StateT (RDF rdfImpl) m a
+
+baseUrl :: (Rdf rdfImpl, Monad m) => RdfST rdfImpl m (Maybe BaseUrl)
+baseUrl = RDF.baseUrl <$> get
+
+prefixMappings :: (Rdf rdfImpl, Monad m) => RdfST rdfImpl m PrefixMappings
+prefixMappings = RDF.prefixMappings <$> get
+
+addPrefixMappings
+  :: (Rdf rdfImpl, Monad m)
+  => PrefixMappings
+  -> Bool
```

Comment:
I know this `Bool` maps to the underlying API but I would opt for distinct functions: one  to add to existing `PrefixMappings` (this function) and another to replace the existing ones, say, `replacePrefixMappings`.

Reply:
@koslambrou This suggestion by @tmciver sounds good to me. Thoughts? Either way, haddock documentation to make the functionality clear would be great.

Reply:
Agreed! I'll change that

Reply:
Done. Also added a comment for each function pointing to the equivalent function in `Data.RDF.Types`.